packages/compass-query-bar/src/query/leniently-fix-query.ts

```diff
@@ -0,0 +1,51 @@
+import detect from 'mongodb-query-parser';
+
+export type CursorPlacement = number | undefined;
+
+const _detect = detect as unknown as (query: string) => boolean;
+function _isValidQuery(query: string): boolean {
+  try {
+    return !!_detect(query);
+  } catch (ex) {
+    return false;
+  }
+}
+
+export function lenientlyFixQuery(query: string): string | false {
+  query = query.trim();
+  let modified = false;
+
+  if (query === '') {
+    return '{${}}';
+  }
+
+  const isValid = _isValidQuery(query);
+
+  if (!isValid) {
+    if (query.startsWith('{') && query.endsWith('}')) {
+      const queryWithoutWrappingBraces = query.substring(1, query.length - 1);
+      const isInnerQueryValid = _isValidQuery(queryWithoutWrappingBraces);
+      if (isInnerQueryValid) {
+        modified = true;
+        query = queryWithoutWrappingBraces;
+      }
+    } else {
+      const wrappedQuery = `{${query}}`;
+      if (_isValidQuery(wrappedQuery)) {
+        modified = true;
+        query = wrappedQuery;
+      }
+    }
+  }
+
+  if (modified) {
+    const caretPosition = query.lastIndexOf('}');
```

Comment:
We probably want to escape here just so we don't accidentally create snippet anchors from user input (here's a [codemirror doc](https://codemirror.net/docs/ref/#autocomplete.snippet) that mentions escaping for snippets)

```suggestion
    query = query.replaceAll('{', '\\{');
    const caretPosition = query.lastIndexOf('}');
```

Reply:
That sounds good, because I was having sometimes behaviour where Codemirror was selecting everything, instead of just placing the cursor.

I'll do the change and test it.